### PR TITLE
Add secondary, smaller Character Count Warning for dialogue lines

### PR DESCRIPTION
**Pull request by gingeranyhow**
Adds a second character count for dialogue vs. narrative lines. Applies dialogue check to dialogue lines (determined by regex). 

For dialogue lines, changes styling of the character name for emphasis.

screen shot 2018-01-15 at 9 29 16 pm

### CODE REVIEW

File: app/renderer/playerView.js [hunk 1]
```diff
@@ -107,41 +107,59 @@ function addTextSection(text)
...
     // If all we have left is a character name, style that like a tag
+
     if (/[A-Z]+:\s*$/.test(nonTagText)) {
         tagText = nonTagText + tagText;
         nonTagText = "";
     }
 
-    var splitIntoSpans = nonTagText.split(" ");
+    var arrayOfWords = nonTagText.split(" ");
+    var dialogueRegex = /^[A-Z]+:/
```
Comment (hoverbird) [inline app/renderer/playerView.js:125]: Worth noting that, while more restrictive than the characterNamePattern Regex we use in C#, all of our current Ink files are correctly matched by this one... So there's something to be said for sticking with something that's simpler and works. I took a look at the C# version and I think the main difference is that it requires that only the very first character to be an uppercase alphanumeric... thereafter a single space or lowercase letter are allowed. The original idea was to allow for names like `McNASTY:` and `JULIUS CAESAR` but so far we don't have any names like this, and they're no longer displayed onscreen, making the point rather moot IMO.

File: app/renderer/playerView.js [hunk 2]
```diff
@@ -107,41 +107,59 @@ function addTextSection(text)
...
+        const MAX_DIALOGUE_COUNT = settings.getSync("dialogCountDanger");
+        let dangerousWordCount = isDialogueLine ? MAX_DIALOGUE_COUNT : MAX_NARRATION_COUNT;
+
+        for (var i = 0; i < arrayOfWords.length; i++) {
+          var currentWord = arrayOfWords[i];
+          charCount += currentWord.length;
+          if (charCount < dangerousWordCount) {
+              plainSpans.push(currentWord);
+          } else {
+            highlightedSpans.push(currentWord);
```
Comment (hoverbird) [inline app/renderer/playerView.js]: inconsistent indentation

File: app/renderer/playerView.js [hunk 3]
```diff
@@ -107,41 +107,59 @@ function addTextSection(text)
...
-            }
+        const MAX_NARRATION_COUNT = settings.getSync("narrativeCountDanger");
+        const MAX_DIALOGUE_COUNT = settings.getSync("dialogCountDanger");
+        let dangerousWordCount = isDialogueLine ? MAX_DIALOGUE_COUNT : MAX_NARRATION_COUNT;
+
+        for (var i = 0; i < arrayOfWords.length; i++) {
+          var currentWord = arrayOfWords[i];
+          charCount += currentWord.length;
+          if (charCount < dangerousWordCount) {
+              plainSpans.push(currentWord);
```
Comment (hoverbird) [inline app/renderer/playerView.js]: We're using two spaces for indentation on this project— should be something you can set your editor to do automatically for you!

File: app/renderer/playerView.js [hunk 4]
```diff
@@ -107,41 +107,59 @@ function addTextSection(text)
...
-        plainSpans = splitIntoSpans;
+        plainSpans = arrayOfWords;
     }
 
-    var textAsSpans = "<span>" + plainSpans.join("</span> <span>") + "</span>";
+    let isLina = (arrayOfWords[0] === 'LINA:');
+    let initialSpan = "<span>";
+    
+    if (isDialogueLine && isLina) {
+      initialSpan = "<span class='characterName Lina'>‍"
```
Comment (hoverbird) [inline app/renderer/playerView.js:156]: It would be slightly cleaner to do this styling with only the one "lina" class, IMO!

**Review by hoverbird — CHANGES_REQUESTED** [review]
One small suggested change, but otherwise looks good. If Andrew feels good about this style-and-code-wise, #shippit!

File: app/renderer/playerView.js (hunk 4 above)
Comment (gingeranyhow) [reply app/renderer/playerView.js:156]: Fixed - removed not lina.

File: app/renderer/playerView.js (hunk 2 above)
Comment (gingeranyhow) [reply app/renderer/playerView.js]: fixed.

**Pull request merged**